WebContent/view/AddOrEditBook.jsp

```diff
@@ -0,0 +1,84 @@
+<%@ include file="../common/import.jsp"%>
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="UTF-8">
+<link href="../static/vendor/bootstrap/css/bootstrap.min.css"
+	rel="stylesheet" type="text/css">
+<link href="../static/css/Style.css" rel="stylesheet" type="text/css">
+<title>Add or Edit Book</title>
+</head>
+<body>
+	<%
+		Book book = (Book) request.getAttribute("book");
+		String url = "/Book/Add";
+		List<Map<String, String>> bookErr = (List) request.getAttribute("bookErr");
+
+		if (book != null) {
+
+			if(book.getBookId() == 0){
```

Comment:
There are 2 conditions here but we need only 1. Let remove which one is unnecessary!

Reply:
I think I should use 2 conditions here, because when there is an error the page will back to the book-form, and the data that the user entered I save in a Book object. That Book hasn't had the Id yet.